public/app/features/dashboard/containers/DashboardPage.tsx

```diff
@@ -465,41 +387,13 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
               <SubMenu dashboard={dashboard} annotations={dashboard.annotations.list} links={dashboard.links} />
             </section>
           )}
-          {config.featureToggles.editPanelCSVDragAndDrop ? (
-            <DropZone
-              onDrop={this.onFileDrop}
-              accept={DFImport.acceptedFiles}
-              maxSize={DFImport.maxFileSize}
-              noClick={true}
-            >
-              {({ getRootProps, isDragActive }) => {
-                const styles = getStyles(this.props.theme, isDragActive);
-                return (
-                  <div {...getRootProps({ className: styles.dropZone })}>
-                    <div className={styles.dropOverlay}>
-                      <div className={styles.dropHint}>
-                        <Icon name="upload" size="xxxl"></Icon>
-                        <h3>Create tables from spreadsheets</h3>
-                      </div>
-                    </div>
-                    <DashboardGrid
-                      dashboard={dashboard}
-                      isEditable={!!dashboard.meta.canEdit}
-                      viewPanel={viewPanel}
-                      editPanel={editPanel}
-                    />
-                  </div>
-                );
-              }}
-            </DropZone>
-          ) : (
-            <DashboardGrid
-              dashboard={dashboard}
-              isEditable={!!dashboard.meta.canEdit}
-              viewPanel={viewPanel}
-              editPanel={editPanel}
-            />
-          )}
```

Comment:
I agree this is a complexity we don't want/need to have in a core component. However, removing the feature is too radical for me.

Instead, we could move all the logic of this component into a new component that will enable the feature if the flag is enabled making sure it doesn't generate any performance issue:
```jsx
<DashboardDropZone dashboard={this.props.dashboard}>
  <DashboardGrid
    dashboard={dashboard}
    isEditable={!!dashboard.meta.canEdit}
    viewPanel={viewPanel}
    editPanel={editPanel}
    />
</DashboardDropZone>

// ...
function DashboardDropZone({dashboard, children}){
  if(noFFEnabled){return children;}
  const onFileDrop = ()=>{...}
  return (
    <DropZone
      onDrop={this.onFileDrop}
      accept={DFImport.acceptedFiles}
      maxSize={DFImport.maxFileSize}
      noClick={true}
      >
      ...
        {children}
      ...
    </DropZone>
  )
}

Reply:
This is not relevant since the feature can be removed